x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/support/WatcherIndexTemplateRegistry.java

```diff
@@ -108,18 +110,23 @@ protected String getOrigin() {
     }
 
     public static boolean validate(ClusterState state) {
+        final Stream<String> watcherHistoryTemplateIds;
         if (state.nodes().getMinNodeVersion().onOrAfter(Version.V_7_9_0)){
-            return (state.getMetadata().templatesV2().containsKey(WatcherIndexTemplateRegistryField.HISTORY_TEMPLATE_NAME) ||
-                state.getMetadata().templatesV2().containsKey(WatcherIndexTemplateRegistryField.HISTORY_TEMPLATE_NAME_NO_ILM));
+            watcherHistoryTemplateIds = state.getMetadata().templatesV2().keySet().stream();
+        } else {
+            watcherHistoryTemplateIds = Arrays.stream(state.getMetadata().getTemplates().keys().toArray(String.class));
         }
+        return watcherHistoryTemplateIds.filter(s -> s.startsWith(".watch-history-"))
+            .map(s -> Integer.valueOf(s.substring(s.lastIndexOf('-') + 1)))
+            .anyMatch(version -> version >= 6);
```

Comment:
Should this be `9` and not `6`? Last `6.x` uses template version `9`, right?

Reply:
Oops, yes version 9 is the version used in 6.8.latest, I misread the `WatcherIndexTemplateRegistryField` in the 6.8 branch. Although looking at the history I don't think even breaking changes were introduced from template version 6.
In anyway I will update this version to 9, since we only support rolling upgrades from 6.8.x to 7.x.